daphne_worker/src/dap.rs

```diff
@@ -64,29 +65,90 @@ pub(crate) fn dap_response_to_worker(resp: DapResponse) -> Result<Response> {
     Ok(worker_resp)
 }
 
+#[async_trait(?Send)]
 impl<D> HpkeDecrypter for DaphneWorkerConfig<D> {
-    fn get_hpke_config_for(&self, _task_id: &Id) -> Option<&HpkeConfig> {
-        if self.hpke_receiver_config_list.is_empty() {
-            return None;
-        }
+    async fn get_hpke_config_for(
+        &self,
+        _task_id: &Id,
+    ) -> std::result::Result<Option<HpkeConfig>, DapError> {
+        let kv_store = self.kv("DAP_HPKE_RECEIVER_CONFIG_STORE").map_err(dap_err)?;
+        let keys = kv_store
```

Comment:
Since we're only ever going to read the first key, it's a good idea to set a limit to the number of keys returned: https://docs.rs/worker-kv/latest/worker_kv/struct.ListOptionsBuilder.html#

While at it, it's probably a good idea to specify a prefix for the query (i.e., "hpke_receiver_config/") so that, if we ever have keys with a different key/value structure in the future ("count" or something), we avoid any bugs that would result.

Reply:
Great idea, I've missed these functions.